client/session.go

```diff
@@ -431,9 +431,7 @@ func (s *session) Write(namespace string, id string, t time.Time, value float64,
 		if err != nil {
 			atomic.AddInt32(&resultErrs, 1)
```

Comment:
What's the point of this atomic integral (`resultErrs`, `pending`, etc) if you immediately grab a lock and extend a slice after that? A simple addition under the lock won't change anything perf-wise but looks cleaner. Disclaimer: I'm not super-familiar with the codebase yet so I might as well just be missing something obvious.